deployment/terraform/iam.tf

```diff
@@ -0,0 +1,82 @@
+#
+# Container Instance IAM resources
+#
+data "aws_iam_policy_document" "container_instance_ec2_assume_role" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["ec2.amazonaws.com"]
+    }
+
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+resource "aws_iam_role" "container_instance_ec2" {
+  name               = "${var.environment}ContainerInstanceProfile"
+  assume_role_policy = data.aws_iam_policy_document.container_instance_ec2_assume_role.json
+}
+
+resource "aws_iam_role_policy_attachment" "ec2_service_role" {
+  role       = aws_iam_role.container_instance_ec2.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role"
```

Comment:
The other AWS canned policy attachments were made into Terraform variables. I think that it would be good to keep things consistent and do that here as well.